### DISCUSSION TITLE
[Photos - Desktop] Support specific date search including year on desktop apps

### DISCUSSION THREAD

**Discussion by sv3nnie** [ente-io/ente#2821] [Enhancements]
On mobile, you can search for a specific date, including the year, and Ente will only show you photos of that specific date. For example, searching for '22 December 2022' will only show me photos of that specific date:
![image](https://github.com/user-attachments/assets/657839cd-d71b-42db-a3ab-7bd55f26215a)

On desktop, you can search for a specific day and month, for example '22 December', but it doesn't take into account the year. It shows me all memories of 22 December and December, but not specifically 22 December **2022**.
![image](https://github.com/user-attachments/assets/006a2bf1-aabf-45e4-9693-5e46a3067722)

**Comment by mnvr** [comment]
We've made these changes. They're already deployed on the web app and are available in the nightly builds for desktop app too. 

* Added support for searching photos by day of week ("Wednesday") and hour of day ("8 pm").
* Made the search by partial dates exact (it'll search for whichever components are specified, e.g. "July", "20 July", "2021", "20 July 2021")
* It will also handle relative dates ("Last month", "Yesterday"). Note that this is currently limited to English phrases, we'll explore other locales too later.
